firmware/controllers/buttonshift.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file        buttonshift.cpp
+ * @brief       Polls pins for gear changes
+ *
+ * @date Aug 31, 2020
+ * @author David Holdeman, (c) 2020
+ */
+#include "buttonshift.h"
+#include "engine.h"
+
+EXTERN_ENGINE;
+
+ButtonShiftController buttonShiftController;
+
+void ButtonShiftController::init (DECLARE_ENGINE_PARAMETER_SIGNATURE) {
+    // 500 millisecond is maybe a little long?
+    debounceUp.init(500, CONFIG(tcuUpshiftButtonPin), CONFIG(tcuUpshiftButtonPinMode));
+    debounceDown.init(500, CONFIG(tcuDownshiftButtonPin), CONFIG(tcuDownshiftButtonPinMode));
+}
+
+void ButtonShiftController::update() {
+    bool upPinState = false;
+    bool downPinState = false;
+    // Don't read the pin if TCU is not enabled, or the pin is not configured
+    if (CONFIG(tcuUpshiftButtonPin) && CONFIG(tcuEnabled)) {
```

Comment:
should this "ignore if not set" logic get put inside the debouncer itself?

Reply:
Maybe? I suppose there might be some advantage to that.